File: proselint-rs/crates/proselint-checks/src/malapropisms.rs
use proselint_registry::checks::{Check, types::*, Padding};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &["Found in the Infinitesimal Universe."];

const CHECK: Check = Check {
	check_type: &Existence {
		items: &[
			"the infinitesimal universe",
			"a serial experience",
			"attack my voracity",
        ],
		padding: Padding::WordsInText,
		exceptions: &[],
	},
	path: "malapropisms.misc",
	msg: "'{}' is a malapropism.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];

File: proselint/checks/malapropisms.py
"""
Malaproprisms.

---
layout:     post
source:     Garner's Modern American Usage
source_url: http://bit.ly/1T4alrY
title:      Malaproprisms
date:       2014-06-10 12:31:19
categories: writing
---

Malapropisms.

"""

from proselint.registry.checks import Check, types

check = Check(
    check_type=types.Existence(
        items=(
            "the infinitesimal universe",
            "a serial experience",
            "attack my voracity",
        )
    ),
    path="malapropisms",
    message="'{}' is a malapropism.",
)

__register__ = (check,)
